external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala

```diff
@@ -1015,6 +1015,65 @@ abstract class AvroSuite
     }
   }
 
+  test("support user provided nullable avro schema " +
+    "for non-nullable catalyst schema without any null record") {
+    withTempPath { tempDir =>
+      val catalystSchema =
+        StructType(Seq(
+          StructField("Age", IntegerType, nullable = false),
+          StructField("Name", StringType, nullable = false)))
+
+      val avroSchema = """
+        |{
+        |  "type" : "record",
+        |  "name" : "test_schema",
+        |  "fields" : [
+        |    {"name": "Age", "type": ["null", "int"]},
+        |    {"name": "Name", "type": ["null", "string"]}
+        |  ]
+        |}
+      """.stripMargin
+
+      val df = spark.createDataFrame(
+        spark.sparkContext.parallelize(Seq(Row(2, "Aurora"))), catalystSchema)
+
+      df.write.format("avro").option("avroSchema", avroSchema).save(tempDir.getPath)
+      checkAvroSchemaEquals(avroSchema, getAvroSchemaStringFromFiles(tempDir.getPath))
+    }
+  }
+
+  test("unsupported nullable avro type") {
+    withTempPath { tempDir =>
+      val catalystSchema =
+        StructType(Seq(
+          StructField("Age", IntegerType, nullable = false),
+          StructField("Name", StringType, nullable = false)))
+
+      for (unsupportedAvroType <- Seq("""["null", "int", "long"]""", """["int", "long"]""")) {
+        val avroSchema = s"""
+          |{
+          |  "type" : "record",
+          |  "name" : "test_schema",
+          |  "fields" : [
+          |    {"name": "Age", "type": $unsupportedAvroType},
+          |    {"name": "Name", "type": ["null", "string"]}
+          |  ]
+          |}
+        """.stripMargin
+
+        val df = spark.createDataFrame(
+          spark.sparkContext.parallelize(Seq(Row(2, "Aurora"))), catalystSchema)
+
+        val tempSavePath = s"$tempDir/save/${UUID.randomUUID()}"
```

Comment:
nit: here we can write to tempDir as well

Reply:
This is required here so we have different save paths for `["null", "int", "long"]` and `["int", "long"]`.

Reply:
You can put `withTempPath { tempDir =>` inside of the loop. `withTempPath` will generate random path for you.

Reply:
I was following the pattern in the other tests in the same suite. Will change to limit the scope of `withTempPath` in the two newly added tests.